File: src/my_cpp_pkg/src/templates/publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"


class PublisherNode : public rclcpp::Node //TODO (Ex: NumberPublisherNode)
{
public:
    PublisherNode() : Node("[NODE NAME]") //TODO (Ex: number_publisher)
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("[TOPIC NAME]", 10); //TODO (Ex: number)
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500),
                                         std::bind(&PublisherNode::publishFunction, this)); //TODO (Ex: publishHardwareStatus)
        RCLCPP_INFO(this->get_logger(), "Publisher has been started.");
    }

private:
    void publishFunction() //TODO
    {
        // DEFINE FORMAT OF MESSAGE
        auto msg = example_interfaces::msg::String(); //TODO (Ex: String, Int64...)
        
        // ADD DATA TO MESSAGE
        msg.data = std::string("Hi, this is a message from the publisher."); 
        publisher_->publish(msg);
    }

    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/templates/publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import Int64 # Depends on the format of the messages that are published

class PublisherNode(Node): # TODO

    def __init__(self):
        super().__init__("[NAME OF NODE]") # TODO

        self.publisher_ = self.create_publisher(Int64, "number", 10) # Topic: /number, queue size 10
        
        # Define rate of publication
        self.timer_ = self.create_timer(0.5, self.publish_function) 

        self.get_logger().info("[NAME OF NODE] has been started") # TODO

    def publish_function(self):
        # Format of message to be sent to publisher's topic
        msg = Int64()   

        # Content of message
        msg.data = 69   
        self.publisher_.publish(msg)



def main(args=None):
    rclpy.init(args=args)
    node = PublisherNode() # TODO
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()